ui/components/app/assets/asset-list/asset-list.tsx

```diff
@@ -104,32 +124,91 @@ const AssetList = ({ onClickAsset, showTokensLinks }: AssetListProps) => {
   // for EVM assets
   const shouldShowTokensLinks = showTokensLinks ?? isEvm;
 
-  const detectedTokensMultichain = useSelector(
-    getAllDetectedTokensForSelectedAddress,
+  const detectedTokensMultichain: {
+    [key: `0x${string}`]: Token[];
+  } = useSelector(getAllDetectedTokensForSelectedAddress);
+
+  const multichainDetectedTokensLength = Object.keys(
+    detectedTokensMultichain || {},
+  ).reduce(
+    (sum, key) => sum + detectedTokensMultichain[key as `0x${string}`].length,
+    0,
   );
 
-  const totalTokens =
-    process.env.PORTFOLIO_VIEW &&
-    !isTokenNetworkFilterEqualCurrentNetwork &&
-    detectedTokensMultichain
-      ? (Object.values(detectedTokensMultichain).reduce(
-          // @ts-expect-error TS18046: 'tokenArray' is of type 'unknown'
-          (count, tokenArray) => count + tokenArray.length,
-          0,
-        ) as number)
-      : detectedTokens.length;
+  // Add detected tokens to sate
+  useEffect(() => {
+    const importAllDetectedTokens = async () => {
```

Comment:
I think eventually we should do this in the controllers instead.  So that when tokens are detected, they just get added to `allTokens` instead of `allDetectedTokens`.  But that's a bigger change, so I don't mind doing this for now.

Reply:
that was my impression as well 🙏